off-chain/src/Network/Types.purs

```diff
@@ -70,42 +125,146 @@ data StatusResponse
   | SubmitError String
   | NotFound
 
+derive instance Generic ActionStatus _
+
+instance showActionStatus :: Show ActionStatus where
+  show = genericShow
+
+instance encodeAesonActionStatus :: EncodeAeson ActionStatus where
+  encodeAeson = case _ of
+    AskForSignature askForSig -> encodeTagged' "AskForSignature"
+      (encodeAskForSig askForSig)
+    ToBeProcessed i -> encodeTagged' "ToBeProcessed" i
+    ToBeSubmited i -> encodeTagged' "ToBeSubmited" i
+    Processing -> encodeTagged' "Processing" ""
+    RejectedAtChainBuilder reason -> encodeTagged' "RejectedAtChainBuilder"
+      reason
+    RequireNewSignature -> encodeTagged' "RequireNewSignature" ""
```

Comment:
we need to change this in the type to just `RequireSignature Transaction` or equivalent